### ISSUE TITLE
Gentoo Support (Plus incoming fix)

### ISSUE THREAD

**Issue by WavyEbuilder** [ArtixLabs/Niagara#1]
Just wanted to create the issue in case anyone is wondering - I am adding gentoo support, and was wondering whether anyone thought it might be a good idea to implement use flags for gentoo at some point? I will have basic gentoo support in ~30mins - 1hour

**Comment by WavyEbuilder** [comment]
So I was just thinking about it and have an idea - maybe if the detected OS is gentoo, then it refers to a file called use.py which parses use flags?

**Comment by Cobalt-Inferno** [comment]
I'll leave this issue open for Gentoo Related features.
The use flag parser seems like a good idea, though if it were to be implemented, it would probably be better to have some other flags that can be specified as  well (for example avoiding binary files)

**Comment by WavyEbuilder** [comment]
Good idea - for example, the firefox package is probably the package most used as a binary, so it would be cool to create a gentoo file that can be passed - maybe by setting a bool _bin_ to true? Thinking about ~amd64 and licensing, it might be useful to have two more bools: unstable and accept_proprietary_license so they can be automatically be unmasked.

**Comment by Cobalt-Inferno** [comment]
Working with portage will be interesting, as things such as building with LTO and 9999 packages might be hard to implement here. I'll setup some masking options & other useful ones later.

**Comment by WavyEbuilder** [comment]
That'll be perfect. Maybe it might be an idea to support package atoms somehow?

**Comment by Cobalt-Inferno** [comment]
I'd consider it, but the major point of Niagara is to be a "write once, run everywhere" style of program. So for any specified package, there will have to be some correlation to a few more distributions.
I intend to eventually add support for some generic package managers (Nix, Brew, etc) to make this easier on abstract distributions.
